### PR TITLE
Ju/rodrigob/add exceptions

### PR DESCRIPTION
**Pull request by rodri-98**

### CODE REVIEW

File: searchobject/src/main/java/org/jalau/at18/searchobject/controller/response/ErrorResponse.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ *
+ * This software is the confidential and proprieraty information of Jalasoft
+ * ("Confidential Information"). You shall not disclose such Confidential
+ * Information and shall use it only in accordance with the terms of the
+ * Licence agreement you entered into with Jalasoft
+ */
+package org.jalau.at18.searchobject.controller.response;
+
+/**
+ *
```
Comment (saraialvarezrosas) [inline searchobject/src/main/java/org/jalau/at18/searchobject/controller/response/ErrorResponse.java]: Add documentation fot this class

File: searchobject/src/main/java/org/jalau/at18/searchobject/controller/response/FaceDetectionResponse.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ *
+ * This software is the confidential and proprieraty information of Jalasoft
+ * ("Confidential Information"). You shall not disclose such Confidential
+ * Information and shall use it only in accordance with the terms of the
+ * Licence agreement you entered into with Jalasoft
+ */
+package org.jalau.at18.searchobject.controller.response;
+
+/**
+ *
```
Comment (saraialvarezrosas) [inline searchobject/src/main/java/org/jalau/at18/searchobject/controller/response/FaceDetectionResponse.java]: Add documentation for this class

**Review by saraialvarezrosas — COMMENTED** [review]
Please add documentation.

File: searchobject/src/main/java/org/jalau/at18/searchobject/common/exception/EmotionRecognizerException.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright (c) 2022 Jala University.
+ *
+ * This software is the confidential and proprieraty information of Jalasoft
```
Comment (FernandezRobert) [inline searchobject/src/main/java/org/jalau/at18/searchobject/common/exception/EmotionRecognizerException.java]: Typo in `proprietarty` it should be `property`

File: searchobject/src/main/java/org/jalau/at18/searchobject/common/exception/FaceDetectionException.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright (c) 2022 Jala University.
+ *
+ * This software is the confidential and proprieraty information of Jalasoft
```
Comment (FernandezRobert) [inline searchobject/src/main/java/org/jalau/at18/searchobject/common/exception/FaceDetectionException.java]: Typo in `proprietarty` it should be `property`

File: searchobject/src/main/java/org/jalau/at18/searchobject/common/exception/FaceDetectionException.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+ * ("Confidential Information"). You shall not disclose such Confidential
+ * Information and shall use it only in accordance with the terms of the
+ * Licence agreement you entered into with Jalasoft
+ */
+package org.jalau.at18.searchobject.common.exception;
+
+/**
+ * Throws this FaceDetectionException if there is a problem in the fyle or in the type when the faceDetect model is used
```
Comment (FernandezRobert) [inline searchobject/src/main/java/org/jalau/at18/searchobject/common/exception/FaceDetectionException.java]: Typo in **`fyle`** should be **`file`**

File: searchobject/src/main/java/org/jalau/at18/searchobject/common/exception/NotifierTypeException.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright (c) 2022 Jala University.
+ *
+ * This software is the confidential and proprieraty information of Jalasoft
```
Comment (FernandezRobert) [inline searchobject/src/main/java/org/jalau/at18/searchobject/common/exception/NotifierTypeException.java]: Typo in **`proprietarty`** it should be **`property`**

File: searchobject/src/main/java/org/jalau/at18/searchobject/common/exception/ObjectRecognizerException.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright (c) 2022 Jala University.
+ *
+ * This software is the confidential and proprieraty information of Jalasoft
```
Comment (FernandezRobert) [inline searchobject/src/main/java/org/jalau/at18/searchobject/common/exception/ObjectRecognizerException.java]: Typo in **`proprietarty`** it should be **`property`**

File: searchobject/src/main/java/org/jalau/at18/searchobject/controller/endpoint/EmotionRecognitionController.java [hunk 8]
```diff
@@ -1,7 +1,16 @@
+/**
+ * Copyright (c) 2022 Jala University.
+ *
+ * This software is the confidential and proprieraty information of Jalasoft
```
Comment (FernandezRobert) [inline searchobject/src/main/java/org/jalau/at18/searchobject/controller/endpoint/EmotionRecognitionController.java]: Typo in **`proprietarty`** it should be **`property`**

File: searchobject/src/main/java/org/jalau/at18/searchobject/controller/endpoint/ProcessFrameController.java [hunk 9]
```diff
@@ -1,13 +1,21 @@
+/**
+ * Copyright (c) 2022 Jala University.
+ *
+ * This software is the confidential and proprieraty information of Jalasoft
```
Comment (FernandezRobert) [inline searchobject/src/main/java/org/jalau/at18/searchobject/controller/endpoint/ProcessFrameController.java]: Typo in **`proprietarty`** it should be **`property`**

File: searchobject/src/main/java/org/jalau/at18/searchobject/controller/endpoint/ProcessFrameController.java [hunk 10]
```diff
@@ -35,24 +42,33 @@ public class ProcessFrameController {
...
+            UnzipFile unzip = new UnzipFile(path);
+            List<MatchInfo> matchInfos = processFrameService.processFrameAccordingCriteria(unzip.getPath(), searchCriteria,
+                    occurrencyPercentage,
+                    modelObjectRecognizer);
 
-        processMatchService.processMatches(matchInfos, notifierType, recipient);
-        return ResponseEntity.status(HttpStatus.OK).body(matchInfos);
+            processMatchService.processMatches(matchInfos, notifierType, recipient);
+            return ResponseEntity.status(HttpStatus.OK).body(matchInfos);
+        } catch (UnzipFileException e) {
```
Comment (FernandezRobert) [inline searchobject/src/main/java/org/jalau/at18/searchobject/controller/endpoint/ProcessFrameController.java]: From lines 66-72, you have three requests that are returning the same thing. you could probably just do the following collapsing.
`} catch (UnzipFileException | ObjectRecognizerException | NotifierTypeException e) {
            return ResponseEntity.badRequest().body(new ErrorResponse("400",e.getMessage()));
        }`

File: searchobject/src/main/java/org/jalau/at18/searchobject/controller/response/ErrorResponse.java [hunk 11]
```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) 2022 Jala University.
+ *
+ * This software is the confidential and proprieraty information of Jalasoft
```
Comment (FernandezRobert) [inline searchobject/src/main/java/org/jalau/at18/searchobject/controller/response/ErrorResponse.java]: Please the typo in `propierty`

File: searchobject/src/main/java/org/jalau/at18/searchobject/controller/response/FaceDetectionResponse.java [hunk 12]
```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (c) 2022 Jala University.
+ *
+ * This software is the confidential and proprieraty information of Jalasoft
```
Comment (FernandezRobert) [inline searchobject/src/main/java/org/jalau/at18/searchobject/controller/response/FaceDetectionResponse.java]: Typo in `proprieraty`

File: searchobject/src/main/java/org/jalau/at18/searchobject/controller/response/ResponseMessage.java [hunk 13]
```diff
@@ -1,4 +1,4 @@
-package org.jalau.at18.searchobject.model.response;
+package org.jalau.at18.searchobject.controller.response;
 
 // The ResponseMessage is for message to client that we’re gonna use in Rest Controller and Exception Handler.
```
Comment (FernandezRobert) [inline searchobject/src/main/java/org/jalau/at18/searchobject/controller/response/ResponseMessage.java]: the word gonna is not so formal to say, you could probably use going to

File: searchobject/src/main/java/org/jalau/at18/searchobject/controller/service/EmotionRecognitionService.java [hunk 14]
```diff
@@ -1,18 +1,29 @@
+/**
+ * Copyright (c) 2022 Jala University.
+ *
+ * This software is the confidential and proprieraty information of Jalasoft
```
Comment (FernandezRobert) [inline searchobject/src/main/java/org/jalau/at18/searchobject/controller/service/EmotionRecognitionService.java]: Typo in `proprieraty`

File: searchobject/src/main/java/org/jalau/at18/searchobject/controller/service/NotifyService.java [hunk 15]
```diff
@@ -1,15 +1,29 @@
...
 import org.jalau.at18.searchobject.notify.Notifier;
 import org.jalau.at18.searchobject.notify.VerifyNotifier;
 import org.springframework.stereotype.Service;
 
 import java.util.List;
 
+/**
+ *
+ *
+ * @throws NotifierTypeExeption if the notifier type is not one of the available notifiers
```
Comment (FernandezRobert) [inline searchobject/src/main/java/org/jalau/at18/searchobject/controller/service/NotifyService.java]: you probably meant `NotifierTypeException` referring to the exception on line 26.

File: searchobject/src/main/java/org/jalau/at18/searchobject/controller/service/NotifyService.java [hunk 16]
```diff
@@ -1,15 +1,29 @@
+/**
+ * Copyright (c) 2022 Jala University.
+ *
+ * This software is the confidential and proprieraty information of Jalasoft
```
Comment (FernandezRobert) [inline searchobject/src/main/java/org/jalau/at18/searchobject/controller/service/NotifyService.java]: Typo in `proprieraty`

File: searchobject/src/main/java/org/jalau/at18/searchobject/controller/service/ProcessFrameService.java [hunk 17]
```diff
@@ -1,18 +1,33 @@
+/**
+ * Copyright (c) 2022 Jala University.
+ *
+ * This software is the confidential and proprieraty information of Jalasoft
```
Comment (FernandezRobert) [inline searchobject/src/main/java/org/jalau/at18/searchobject/controller/service/ProcessFrameService.java]: Typo

File: searchobject/src/main/java/org/jalau/at18/searchobject/controller/service/ProcessMatchService.java [hunk 18]
```diff
@@ -1,17 +1,31 @@
...
 import org.jalau.at18.searchobject.model.objectrecognizer.recognizertypes.MatchInfo;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 import java.util.List;
 
+/**
+ *
+ *
+ * @throws NotifierTypeExeption if the notifier type is not one of the available notifiers
```
Comment (FernandezRobert) [inline searchobject/src/main/java/org/jalau/at18/searchobject/controller/service/ProcessMatchService.java]: probably you meant `NotifierTypeException` referring to line 28

File: searchobject/src/main/java/org/jalau/at18/searchobject/model/emotionrecognizer/EmotionRecognizer.java [hunk 19]
```diff
@@ -1,7 +1,14 @@
+/**
+ * Copyright (c) 2022 Jala University.
+ *
+ * This software is the confidential and proprieraty information of Jalasoft
```
Comment (FernandezRobert) [inline searchobject/src/main/java/org/jalau/at18/searchobject/model/emotionrecognizer/EmotionRecognizer.java]: Typo

File: searchobject/src/main/java/org/jalau/at18/searchobject/model/emotionrecognizer/EmotionRecognizer.java [hunk 20]
```diff
@@ -11,28 +18,39 @@
 import java.util.Scanner;
 
 import org.apache.commons.io.FileUtils;
+import org.jalau.at18.searchobject.common.exception.EmotionRecognizerException;
+
 import com.google.gson.Gson;
```
Comment (FernandezRobert) [inline searchobject/src/main/java/org/jalau/at18/searchobject/model/emotionrecognizer/EmotionRecognizer.java]: not using this import.

File: searchobject/src/main/java/org/jalau/at18/searchobject/model/emotionrecognizer/EmotionRecognizer.java [hunk 21]
```diff
@@ -11,28 +18,39 @@
...
-        String secondPart = "\"\n},\n\"features\":[\n{\n\"maxResults\":10,\n\"type\":\"FACE_DETECTION\"\n}\n]\n}\n]\n}";
-        String imageBase64 = firstPart + encodedString + secondPart;
+    private String convertImage(String filePath) throws EmotionRecognizerException {
+        try {
+            byte[] fileContent = FileUtils.readFileToByteArray(new File(filePath));
+            String encodedString = Base64.getEncoder().encodeToString(fileContent);
+            String firstPart = "{\n\"requests\":[\n{\n\"image\":{\n\"content\":\"";
+            String secondPart = "\"\n},\n\"features\":[\n{\n\"maxResults\":10,\n\"type\":\"FACE_DETECTION\"\n}\n]\n}\n]\n}";
+            String imageBase64 = firstPart + encodedString + secondPart;
         return imageBase64;
```
Comment (FernandezRobert) [inline searchobject/src/main/java/org/jalau/at18/searchobject/model/emotionrecognizer/EmotionRecognizer.java:49]: Line 49-50 can be put together as `return firstPart + encodedString + secondPart;`

File: searchobject/src/main/java/org/jalau/at18/searchobject/model/facedetector/FaceDetect.java [hunk 22]
```diff
@@ -77,8 +79,7 @@ public static void faceDetection (String file, String type) {
 
             String finalFile = file.substring(22,file.length());
```
Comment (FernandezRobert) [inline searchobject/src/main/java/org/jalau/at18/searchobject/model/facedetector/FaceDetect.java:80]: we can change this line to `String finalFile = file.substring(22);`

File: searchobject/src/main/java/org/jalau/at18/searchobject/model/objectrecognizer/UnzipFile.java [hunk 23]
```diff
@@ -1,19 +1,27 @@
+/**
+ * Copyright (c) 2022 Jala University.
+ *
+ * This software is the confidential and proprieraty information of Jalasoft
```
Comment (FernandezRobert) [inline searchobject/src/main/java/org/jalau/at18/searchobject/model/objectrecognizer/UnzipFile.java]: Typo

File: searchobject/src/main/java/org/jalau/at18/searchobject/model/objectrecognizer/UnzipFile.java [hunk 24]
```diff
@@ -1,19 +1,27 @@
...
 public class UnzipFile {
     Path dir;
-    public UnzipFile(Path path) throws IOException {
+    public UnzipFile(Path path) throws UnzipFileException {
         unzip(path);
     }
-    private void unzip(Path path) throws IOException {
+    private void unzip(Path path) throws UnzipFileException {
+        try {
             Path fileZip = path;
```
Comment (FernandezRobert) [inline searchobject/src/main/java/org/jalau/at18/searchobject/model/objectrecognizer/UnzipFile.java:25]: this line is not necesary, and we can have line 26 like `File destDir = new File(getFolder(path.toString()));`

File: searchobject/src/main/java/org/jalau/at18/searchobject/model/objectrecognizer/recognizer/VerifyModelRecognizer.java [hunk 25]
```diff
@@ -1,9 +1,24 @@
+/**
+ * Copyright (c) 2022 Jala University.
+ *
+ * This software is the confidential and proprieraty information of Jalasoft
```
Comment (FernandezRobert) [inline searchobject/src/main/java/org/jalau/at18/searchobject/model/objectrecognizer/recognizer/VerifyModelRecognizer.java]: Typo

File: searchobject/src/main/java/org/jalau/at18/searchobject/notify/VerifyNotifier.java [hunk 26]
```diff
@@ -1,10 +1,25 @@
...
+ * Licence agreement you entered into with Jalasoft
+ */
 package org.jalau.at18.searchobject.notify;
 
+import org.jalau.at18.searchobject.common.exception.NotifierTypeException;
+
+/**
+ *
+ *
+ * @throws NotifierTypeExeption if the notifier type is not one of the available notifiers
```
Comment (FernandezRobert) [inline searchobject/src/main/java/org/jalau/at18/searchobject/notify/VerifyNotifier.java]: probably you meant `NotifierTypeException`

**Review by FernandezRobert — CHANGES_REQUESTED** [review]

File: searchobject/src/main/java/org/jalau/at18/searchobject/controller/response/ErrorResponse.java (hunk 1 above)
Comment (rodri-98) [reply searchobject/src/main/java/org/jalau/at18/searchobject/controller/response/ErrorResponse.java:12]: It's already added

File: searchobject/src/main/java/org/jalau/at18/searchobject/controller/response/FaceDetectionResponse.java (hunk 2 above)
Comment (rodri-98) [reply searchobject/src/main/java/org/jalau/at18/searchobject/controller/response/FaceDetectionResponse.java]: It's already added

File: searchobject/src/main/java/org/jalau/at18/searchobject/controller/response/ErrorResponse.java (hunk 1 above)
Comment (rodri-98) [reply searchobject/src/main/java/org/jalau/at18/searchobject/controller/response/ErrorResponse.java]: It's already added

File: searchobject/src/main/java/org/jalau/at18/searchobject/common/exception/EmotionRecognizerException.java (hunk 3 above)
Comment (rodri-98) [reply searchobject/src/main/java/org/jalau/at18/searchobject/common/exception/EmotionRecognizerException.java]: It's already changed

File: searchobject/src/main/java/org/jalau/at18/searchobject/common/exception/FaceDetectionException.java (hunk 4 above)
Comment (rodri-98) [reply searchobject/src/main/java/org/jalau/at18/searchobject/common/exception/FaceDetectionException.java]: It's already changed

File: searchobject/src/main/java/org/jalau/at18/searchobject/common/exception/FaceDetectionException.java (hunk 5 above)
Comment (rodri-98) [reply searchobject/src/main/java/org/jalau/at18/searchobject/common/exception/FaceDetectionException.java]: It's already changed

File: searchobject/src/main/java/org/jalau/at18/searchobject/common/exception/NotifierTypeException.java (hunk 6 above)
Comment (rodri-98) [reply searchobject/src/main/java/org/jalau/at18/searchobject/common/exception/NotifierTypeException.java]: It's already changed

File: searchobject/src/main/java/org/jalau/at18/searchobject/common/exception/ObjectRecognizerException.java (hunk 7 above)
Comment (rodri-98) [reply searchobject/src/main/java/org/jalau/at18/searchobject/common/exception/ObjectRecognizerException.java]: It's already changed

File: searchobject/src/main/java/org/jalau/at18/searchobject/controller/endpoint/EmotionRecognitionController.java (hunk 8 above)
Comment (rodri-98) [reply searchobject/src/main/java/org/jalau/at18/searchobject/controller/endpoint/EmotionRecognitionController.java]: It's already changed

File: searchobject/src/main/java/org/jalau/at18/searchobject/controller/endpoint/ProcessFrameController.java (hunk 9 above)
Comment (rodri-98) [reply searchobject/src/main/java/org/jalau/at18/searchobject/controller/endpoint/ProcessFrameController.java]: It's already changed

File: searchobject/src/main/java/org/jalau/at18/searchobject/controller/endpoint/ProcessFrameController.java (hunk 10 above)
Comment (rodri-98) [reply searchobject/src/main/java/org/jalau/at18/searchobject/controller/endpoint/ProcessFrameController.java]: It's already changed

File: searchobject/src/main/java/org/jalau/at18/searchobject/controller/response/ErrorResponse.java (hunk 11 above)
Comment (rodri-98) [reply searchobject/src/main/java/org/jalau/at18/searchobject/controller/response/ErrorResponse.java]: It's already changed

File: searchobject/src/main/java/org/jalau/at18/searchobject/controller/response/FaceDetectionResponse.java (hunk 12 above)
Comment (rodri-98) [reply searchobject/src/main/java/org/jalau/at18/searchobject/controller/response/FaceDetectionResponse.java]: It's already changed

File: searchobject/src/main/java/org/jalau/at18/searchobject/controller/response/ResponseMessage.java (hunk 13 above)
Comment (rodri-98) [reply searchobject/src/main/java/org/jalau/at18/searchobject/controller/response/ResponseMessage.java]: It's already changed

File: searchobject/src/main/java/org/jalau/at18/searchobject/controller/service/EmotionRecognitionService.java (hunk 14 above)
Comment (rodri-98) [reply searchobject/src/main/java/org/jalau/at18/searchobject/controller/service/EmotionRecognitionService.java]: It's already changed

File: searchobject/src/main/java/org/jalau/at18/searchobject/controller/service/NotifyService.java (hunk 15 above)
Comment (rodri-98) [reply searchobject/src/main/java/org/jalau/at18/searchobject/controller/service/NotifyService.java]: It's already changed

File: searchobject/src/main/java/org/jalau/at18/searchobject/controller/service/NotifyService.java (hunk 16 above)
Comment (rodri-98) [reply searchobject/src/main/java/org/jalau/at18/searchobject/controller/service/NotifyService.java]: It's already changed

File: searchobject/src/main/java/org/jalau/at18/searchobject/controller/service/ProcessFrameService.java (hunk 17 above)
Comment (rodri-98) [reply searchobject/src/main/java/org/jalau/at18/searchobject/controller/service/ProcessFrameService.java]: It's already changed

File: searchobject/src/main/java/org/jalau/at18/searchobject/controller/service/ProcessMatchService.java (hunk 18 above)
Comment (rodri-98) [reply searchobject/src/main/java/org/jalau/at18/searchobject/controller/service/ProcessMatchService.java]: It's already changed

File: searchobject/src/main/java/org/jalau/at18/searchobject/model/emotionrecognizer/EmotionRecognizer.java (hunk 19 above)
Comment (rodri-98) [reply searchobject/src/main/java/org/jalau/at18/searchobject/model/emotionrecognizer/EmotionRecognizer.java]: It's already changed

File: searchobject/src/main/java/org/jalau/at18/searchobject/model/emotionrecognizer/EmotionRecognizer.java (hunk 20 above)
Comment (rodri-98) [reply searchobject/src/main/java/org/jalau/at18/searchobject/model/emotionrecognizer/EmotionRecognizer.java]: It's already changed

File: searchobject/src/main/java/org/jalau/at18/searchobject/model/objectrecognizer/UnzipFile.java (hunk 23 above)
Comment (rodri-98) [reply searchobject/src/main/java/org/jalau/at18/searchobject/model/objectrecognizer/UnzipFile.java]: It's already changed

File: searchobject/src/main/java/org/jalau/at18/searchobject/model/objectrecognizer/recognizer/VerifyModelRecognizer.java (hunk 25 above)
Comment (rodri-98) [reply searchobject/src/main/java/org/jalau/at18/searchobject/model/objectrecognizer/recognizer/VerifyModelRecognizer.java]: It's already changed

File: searchobject/src/main/java/org/jalau/at18/searchobject/notify/VerifyNotifier.java (hunk 26 above)
Comment (rodri-98) [reply searchobject/src/main/java/org/jalau/at18/searchobject/notify/VerifyNotifier.java]: It's already changed

**Review by FernandezRobert — APPROVED** [review]
Nice job!

**Review by saraialvarezrosas — APPROVED** [review]

**Review by DaniMB95 — APPROVED** [review]

**Pull request merged**